vg/vgtex/canvas.go

```diff
@@ -0,0 +1,296 @@
+// Copyright 2016 The gonum Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package vgtex provides a vg.Canvas implementation for LaTeX, targeted at the TikZ/PGF LaTeX package:
+//  https://sourceforge.net/projects/pgf
+//
+// vgtex generates PGF instructions that will be interpreted and rendered by LaTeX.
+// vgtex allows to put any valid LaTeX notation inside plot's strings.
+package vgtex
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"image/color"
+	"io"
+	"math"
+	"strings"
+
+	"github.com/gonum/plot/vg"
+)
+
+const degPerRadian = 180 / math.Pi
+
+const (
+	defaultHeader = `%%%%%% generated by gonum/plot %%%%%%
+\documentclass{standalone}
+\usepackage{pgf}
+\begin{document}
+`
+	defaultFooter = "\\end{document}\n"
+)
+
+// Canvas implements the vg.Canvas interface, translating gonum/plot
+// drawing primitives into PGF ones.
+type Canvas struct {
+	buf   *bytes.Buffer
+	w, h  vg.Length
+	dpi   int // dpi is the number of dots per inch for this canvas
+	stack []context
+
+	// if document is true, Canvas.WriteTo will generate a standalone
+	// .tex file that can be fed to, e.g., pdflatex.
+	document bool
+
+	Header string // Header is prepended to the .tex file when in Document mode
```

Comment:
I guess that I assumed that the `New` and `NewDocument` approach wouldn't have these sometimes-not-used exported fields. Can we just have a fixed default? If someone's creating a document to view a single plot, do they ever care to change the header?